File: OrderManagementService/Models/OrderItem.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;

namespace OrderManagementService.Models

{
    public class OrderItem
    {
        [BsonElement("productId")]
        public string ProductId { get; set; }

        [BsonElement("productVariantId")]
        public string ProductVariantId { get; set; }

        [BsonElement("productName")]
        public string ProductName { get; set; }

        [BsonElement("variantName")]
        public string VariantName { get; set; }

        [BsonElement("quantity")]
        public int Quantity { get; set; }

        [BsonElement("price")]
        public decimal Price { get; set; }

        [BsonElement("productItemIds")]
        public List<string> ProductItemIds { get; set; } = new();
    }

}

File: frontend/lib/models/OrderItem.dart
// Represents a single order item (product + optional variant).
class OrderItem {
  final String productId;
  final String? productVariantId;
  final String productName;
  final String variantName;
  final int quantity;
  final double price;
  final String? imageUrl; // Added field for the product image

  OrderItem({
    required this.productId,
    this.productVariantId,
    required this.productName,
    required this.variantName,
    required this.quantity,
    required this.price,
    this.imageUrl, // Make it optional in the constructor
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => OrderItem(
    productId: json['productId'] as String,
    productVariantId: json['productVariantId'] as String?,
    productName: json['productName'] as String,
    variantName: json['variantName'] as String,
    quantity: json['quantity'] as int,
    price: (json['price'] as num).toDouble(),
    imageUrl: json['imageUrl'] as String?, // Parse imageUrl from JSON
  );

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{
      'productId': productId,
      'productName': productName,
      'variantName': variantName,
      'quantity': quantity,
      'price': price,
    };
    if (productVariantId != null) {
      map['productVariantId'] = productVariantId;
    }
    if (imageUrl != null) {
      map['imageUrl'] = imageUrl; // Include imageUrl in JSON serialization
    }
    return map;
  }
}